start.py

```diff
@@ -0,0 +1,121 @@
+                                                             args.operating_system or DEFAULT_OPERATING_SYSTEM)
+    kdc_hostname = args.kdc_node[0]
+    kdc_node = Node(hostname=kdc_hostname, group='kdc', image=kdc_image,
+                    volumes=[{kerberos_volume_dir: KERBEROS_VOLUME_DIR}])
+    cluster = Cluster(kdc_node, *nodes)
+    cluster.start(args.network)
+
+    logger.info('Updating KDC configurations ...')
+    realm = cluster.network.upper()
+    # Update configurations
+    krb5_conf_data = kdc_node.get_file(KDC_KRB5_CONF_FILENAME)
+    kdc_node.put_file(KDC_KRB5_CONF_FILENAME,
+                      re.sub(r'EXAMPLE.COM', realm,
+                             re.sub(r'example.com', cluster.network,
+                                    re.sub(r'kerberos.example.com', r'{}.{}'.format(kdc_hostname, cluster.network),
+                                           krb5_conf_data))))
+    kdc_conf_data = kdc_node.get_file(KDC_CONF_FILENAME)
+    kdc_node.put_file(KDC_CONF_FILENAME,
+                      re.sub(r'EXAMPLE.COM', realm,
+                             kdc_conf_data.replace(r'[kdcdefaults]',
+                                                   '[kdcdefaults]\n max_renewablelife = 7d\n max_life = 1d')))
+    acl_data = kdc_node.get_file(KDC_ACL_FILENAME)
+    kdc_node.put_file(KDC_ACL_FILENAME, re.sub(r'EXAMPLE.COM', realm, acl_data))
+
+    kdc_commands = [
+        'kdb5_util create -s -r {realm} -P kdcadmin'.format(realm=realm),
+        'kadmin.local -q "addprinc -pw {adminpw} admin/admin@{realm}"'.format(adminpw='acladmin', realm=realm)
+    ]
+
+    logger.info('Starting KDC ...')
+    # Add the following commands before starting kadmin daemon etc.
+    if args.kerberos_principals:
+        principal_list = ['{}@{}'.format(primary, realm) for primary in args.kerberos_principals.split(',')]
+        create_principals_cmds = ['kadmin.local -q "addprinc -randkey {}"'.format(principal)
+                                  for principal in principal_list]
+        kdc_commands.extend(create_principals_cmds)
+
+        create_keytab_cmd = 'kadmin.local -q "xst -norandkey -k {} {}" '.format(KDC_KEYTAB_FILENAME,
+                                                                                ' '.join(principal_list))
+        kdc_commands.append(create_keytab_cmd)
+
+    kdc_commands.extend([
+        'krb5kdc',
+        'kadmind',
+        'authconfig --enablekrb5 --update',
+        'service sshd start',
+        'service krb5kdc start',
+        'service kadmin start'
+    ])
+
+    # Gather keytab file and krb5.conf file in KERBEROS_VOLUME_DIR directory which is mounted on host.
+    kdc_commands.append('cp {} {}'.format(KDC_KRB5_CONF_FILENAME, KERBEROS_VOLUME_DIR))
+    if args.kerberos_principals:
+        kdc_commands.append('chmod 644 {}'.format(KDC_KEYTAB_FILENAME))
+
+    kdc_node.execute(command="bash -c '{}'".format('; '.join(kdc_commands)), quiet=True)
```

Comment:
quiet should be set to `not args.verbose` to allow users to see command output. Same applies to any place we run `execute`

Reply:
Done